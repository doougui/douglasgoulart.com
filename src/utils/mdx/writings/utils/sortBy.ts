import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
