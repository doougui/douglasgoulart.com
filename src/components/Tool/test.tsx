import { render } from 'utils/tests';
import toolMock from './mock';

import { Tool } from '.';

describe('<Tool />', () => {
  it('should render correctly', () => {
    const { container } = render(<Tool {...toolMock} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
