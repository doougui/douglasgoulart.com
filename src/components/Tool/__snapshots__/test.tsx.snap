// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tool /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
}

.c1 {
  width: 8px;
  height: 8px;
  border-radius: 50%;
  background-color: #2D79C7;
}

.c2 {
  font-size: 1.6rem;
  color: var(--text);
}

<div
  class="c0"
>
  <div
    class="c1"
    color="#2D79C7"
  />
  <span
    class="c2"
  >
    TypeScript
  </span>
</div>
`;
